File: leetcode solutions/C++/house robber/houseRobber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {

        if (nums.size() == 0)
                return 0;

        int value1 = nums[0];

        if (nums.size() == 1)
            return value1;

        int value2 = max(value1, nums[1]);

        if (nums.size() == 2)
            return max(value1, value2);

        int value = 0;

        for (int i = 2; i < nums.size(); i++) {
            value = max(value2, value1 + nums[i]);
            value1 = value2;
            value2 = value;
        }
        return value;
    }
};

File: leetcode solutions/Python/house robber/houseRobber.py
class Solution(object):
    def rob(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0

        value1 = nums[0]

        if len(nums) == 1:
            return value1

        value2  = max(value1, nums[1])

        if len(nums) == 2:
            return max(value1, value2)

        for i in xrange(2, len(nums)):
            value = max(value2, value1 + nums[i])
            value1 = value2
            value2 = value

        return value

